### PR TITLE
[new release] alcotest, alcotest-async, alcotest-mirage and alcotest-lwt (1.4.0)

### PR DESCRIPTION
**Pull request by CraigFe**
Alcotest is a lightweight and colourful test framework

- Project page: <a href="https://github.com/mirage/alcotest">https://github.com/mirage/alcotest</a>
- Documentation: <a href="https://mirage.github.io/alcotest">https://mirage.github.io/alcotest</a>

##### CHANGES:

- Add `?here` and `?pos` arguments to the test assertion functions. These can be
  used to pass information about the location of the call-site, which is
  displayed in failing test output. (mirage/alcotest#291, @CraigFe)

- Add a pretty-printer for the exception raised by `Alcotest.check` and related
  functions. This allows them to be used outside of an Alcotest test runner for
  making general assertions.  (mirage/alcotest#296, @CraigFe)

- Add `--bail` option (and corresponding `ALCOTEST_BAIL` environment variable),
  which causes Alcotest to terminate after the first test failure. (mirage/alcotest#298,
  @CraigFe)

### CODE REVIEW

File: packages/alcotest-async/alcotest-async.1.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "core"
+  "base"
+  "async_kernel"
+  "ocaml" {>= "4.03.0"}
+  "alcotest" {= version}
+  "async_unix" {>= "v0.9.0"}
+  "core_kernel" {>= "v0.9.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/alcotest-async/alcotest-async.1.4.0/opam:24]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/alcotest-lwt/alcotest-lwt.1.4.0/opam [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  "re" {with-test}
+  "cmdliner" {with-test}
+  "fmt"
+  "ocaml" {>= "4.03.0"}
+  "alcotest" {= version}
+  "lwt"
+  "logs"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/alcotest-lwt/alcotest-lwt.1.4.0/opam:21]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/alcotest-mirage/alcotest-mirage.1.4.0/opam [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  "fmt"
+  "ocaml" {>= "4.03.0"}
+  "alcotest" {= version}
+  "mirage-clock" {>= "2.0.0"}
+  "duration"
+  "lwt"
+  "logs"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/alcotest-mirage/alcotest-mirage.1.4.0/opam:23]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/alcotest/alcotest.1.4.0/opam [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  "fmt" {>= "0.8.7"}
+  "astring"
+  "cmdliner"
+  "uuidm"
+  "re"
+  "stdlib-shims"
+  "uutf"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/alcotest/alcotest.1.4.0/opam:32]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by CraigFe** [comment]
There are quite a lot of timeouts in the rev-dep tests, but of the ones that actually ran I think I'm satisfied that none of the failures are due to this release.

**Comment by mseri** [comment]
I went through all the errors (I believe). Most of the cancellations are actually successes.
The only failures that may be related are:
```
#=== ERROR while compiling mirage-block-ramdisk.0.4 ===========================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/mirage-block-ramdisk.0.4
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p mirage-block-ramdisk -j 47
# exit-code            1
# env-file             ~/.opam/log/mirage-block-ramdisk-23-e19ae3.env
# output-file          ~/.opam/log/mirage-block-ramdisk-23-e19ae3.out
### output ###
#       ocamlc test/.tests.eobjs/byte/tests.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.tests.eobjs/byte -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/io-page -I /home/opam/.opam/4.12/lib/io-page-unix -I /home/opam/.opam/4.12/lib/logs -I /home/opam/.opam/4.12/lib/lwt -I /home/opam/.opam/4.12/lib/lwt/unix -I /home/opam/.opam/4.12/lib/mirage-block -I /home/opam/.opam/4.12/lib/mirage-block-lwt -I /home/opam/.opam/4.12/lib/mirage-device -I /home/opam/.opam/4.12/lib/mmap -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocplib-endian -I /home/opam/.opam/4.12/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I src/.mirage_block_ramdisk.objs/byte -no-alias-deps -o test/.tests.eobjs/byte/tests.cmo -c -impl test/tests.ml)
# File "test/tests.ml", line 30, characters 25-38:
# 30 |   | Error e -> Fmt.kstrf Alcotest.fail "%a" pp_error e
#                               ^^^^^^^^^^^^^
# Error: This expression has type
#          (string -> 'a) Alcotest.extra_info =
#            ?here:Alcotest.Source_code_position.here ->
#            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
#        but an expression was expected of type string -> 'b
```

```
#=== ERROR while compiling mirage-block-ramdisk.0.5 ===========================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/mirage-block-ramdisk.0.5
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p mirage-block-ramdisk -j 47
# exit-code            1
# env-file             ~/.opam/log/mirage-block-ramdisk-24-323ed7.env
# output-file          ~/.opam/log/mirage-block-ramdisk-24-323ed7.out
### output ###
#       ocamlc test/.tests.eobjs/byte/tests.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.tests.eobjs/byte -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/io-page -I /home/opam/.opam/4.12/lib/io-page-unix -I /home/opam/.opam/4.12/lib/logs -I /home/opam/.opam/4.12/lib/lwt -I /home/opam/.opam/4.12/lib/lwt/unix -I /home/opam/.opam/4.12/lib/mirage-block -I /home/opam/.opam/4.12/lib/mirage-block-combinators -I /home/opam/.opam/4.12/lib/mirage-device -I /home/opam/.opam/4.12/lib/mmap -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocplib-endian -I /home/opam/.opam/4.12/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I src/.mirage_block_ramdisk.objs/byte -no-alias-deps -o test/.tests.eobjs/byte/tests.cmo -c -impl test/tests.ml)
# File "test/tests.ml", line 30, characters 25-38:
# 30 |   | Error e -> Fmt.kstrf Alcotest.fail "%a" pp_error e
#                               ^^^^^^^^^^^^^
# Error: This expression has type
#          (string -> 'a) Alcotest.extra_info =
#            ?here:Alcotest.Source_code_position.here ->
#            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
#        but an expression was expected of type string -> 'b
```

```
#=== ERROR while compiling mirage-channel-lwt.3.1.0 ===========================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/mirage-channel-lwt.3.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build jbuilder runtest -p mirage-channel-lwt -j 47
# exit-code            1
# env-file             ~/.opam/log/mirage-channel-lwt-23-f33fa5.env
# output-file          ~/.opam/log/mirage-channel-lwt-23-f33fa5.out
### output ###
#       ocamlc test/.test.eobjs/test_channel.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.test.eobjs -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/io-page -I /home/opam/.opam/4.12/lib/io-page-unix -I /home/opam/.opam/4.12/lib/logs -I /home/opam/.opam/4.12/lib/lwt -I /home/opam/.opam/4.12/lib/lwt/unix -I /home/opam/.opam/4.12/lib/mirage-channel -I /home/opam/.opam/4.12/lib/mirage-clock -I /home/opam/.opam/4.12/lib/mirage-device -I /home/opam/.opam/4.12/lib/mirage-flow -I /home/opam/.opam/4.12/lib/mirage-flow-lwt -I /home/opam/.opam/4.12/lib/mmap -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocplib-endian -I /home/opam/.opam/4.12/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I lwt/.mirage_channel_lwt.objs -no-alias-deps -o test/.test.eobjs/test_channel.cmo -c -impl test/test_channel.ml)
# File "test/test_channel.ml", line 6, characters 25-38:
# 6 | let fail fmt = Fmt.kstrf Alcotest.fail fmt
#                              ^^^^^^^^^^^^^
# Error: This expression has type
#          (string -> 'a) Alcotest.extra_info =
#            ?here:Alcotest.Source_code_position.here ->
#            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
#        but an expression was expected of type string -> 'b
```

```
#=== ERROR while compiling mirage-channel-lwt.3.2.0 ===========================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/mirage-channel-lwt.3.2.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p mirage-channel-lwt -j 47
# exit-code            1
# env-file             ~/.opam/log/mirage-channel-lwt-23-f33fa5.env
# output-file          ~/.opam/log/mirage-channel-lwt-23-f33fa5.out
### output ###
#       ocamlc test/.test.eobjs/byte/test_channel.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.test.eobjs/byte -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/io-page -I /home/opam/.opam/4.12/lib/io-page-unix -I /home/opam/.opam/4.12/lib/logs -I /home/opam/.opam/4.12/lib/lwt -I /home/opam/.opam/4.12/lib/lwt/unix -I /home/opam/.opam/4.12/lib/mirage-channel -I /home/opam/.opam/4.12/lib/mirage-clock -I /home/opam/.opam/4.12/lib/mirage-device -I /home/opam/.opam/4.12/lib/mirage-flow -I /home/opam/.opam/4.12/lib/mirage-flow-lwt -I /home/opam/.opam/4.12/lib/mmap -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocplib-endian -I /home/opam/.opam/4.12/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I lwt/.mirage_channel_lwt.objs/byte -no-alias-deps -o test/.test.eobjs/byte/test_channel.cmo -c -impl test/test_channel.ml)
# File "test/test_channel.ml", line 5, characters 25-38:
# 5 | let fail fmt = Fmt.kstrf Alcotest.fail fmt
#                              ^^^^^^^^^^^^^
# Error: This expression has type
#          (string -> 'a) Alcotest.extra_info =
#            ?here:Alcotest.Source_code_position.here ->
#            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
#        but an expression was expected of type string -> 'b
```

```
#=== ERROR while compiling mirage-crypto-ec.0.9.2 =============================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/mirage-crypto-ec.0.9.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p mirage-crypto-ec -j 47
# exit-code            1
# env-file             ~/.opam/log/mirage-crypto-ec-23-96bece.env
# output-file          ~/.opam/log/mirage-crypto-ec-23-96bece.out
### output ###
#       ocamlc tests/.test_ec_wycheproof.eobjs/byte/dune__exe__Test_ec_wycheproof.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I tests/.test_ec_wycheproof.eobjs/byte -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/asn1-combinators -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/base/caml -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/biniou -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/easy-format -I /home/opam/.opam/4.12/lib/eqaf -I /home/opam/.opam/4.12/lib/eqaf/bigstring -I /home/opam/.opam/4.12/lib/eqaf/cstruct -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/hex -I /home/opam/.opam/4.12/lib/mirage-crypto -I /home/opam/.opam/4.12/lib/mirage-crypto-pk -I /home/opam/.opam/4.12/lib/mirage-crypto-rng -I /home/opam/.opam/4.12/lib/parsexp -I /home/opam/.opam/4.12/lib/ppx_deriving/runtime -I /home/opam/.opam/4.12/lib/ppx_deriving_yojson/runtime -I /home/opam/.opam/4.12/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.12/lib/ptime -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/rresult -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/sexplib -I /home/opam/.opam/4.12/lib/sexplib0 -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I /home/opam/.opam/4.12/lib/yojson -I /home/opam/.opam/4.12/lib/zarith -I ec/.mirage_crypto_ec.objs/byte -I tests/wycheproof/.wycheproof.objs/byte -no-alias-deps -o tests/.test_ec_wycheproof.eobjs/byte/dune__exe__Test_ec_wycheproof.cmo -c -impl tests/test_ec_wycheproof.ml)
# File "tests/test_ec_wycheproof.ml", line 91, characters 33-46:
# 91 |   | Error err -> Printf.ksprintf Alcotest.fail "While parsing %d: %s" tcId err
#                                       ^^^^^^^^^^^^^
# Error: This expression has type
#          (string -> 'a) Alcotest.extra_info =
#            ?here:Alcotest.Source_code_position.here ->
#            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
#        but an expression was expected of type string -> 'b
```

```
#=== ERROR while compiling tcpip.6.1.0 ========================================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | file:///src
# path                 ~/.opam/4.12/.opam-switch/build/tcpip.6.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p tcpip -j 47
# exit-code            1
# env-file             ~/.opam/log/tcpip-23-088a83.env
# output-file          ~/.opam/log/tcpip-23-088a83.out
### output ###
#       ocamlc test/.test.eobjs/byte/dune__exe__Common.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.test.eobjs/byte -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/arp -I /home/opam/.opam/4.12/lib/arp/mirage -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/cstruct-lwt -I /home/opam/.opam/4.12/lib/domain-name -I /home/opam/.opam/4.12/lib/duration -I /home/opam/.opam/4.12/lib/ethernet -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/ipaddr -I /home/opam/.opam/4.12/lib/ipaddr-cstruct -I /home/opam/.opam/4.12/lib/ipaddr/unix -I /home/opam/.opam/4.12/lib/logs -I /home/opam/.opam/4.12/lib/lru -I /home/opam/.opam/4.12/lib/lwt -I /home/opam/.opam/4.12/lib/lwt-dllist -I /home/opam/.opam/4.12/lib/lwt/unix -I /home/opam/.opam/4.12/lib/macaddr -I /home/opam/.opam/4.12/lib/macaddr-cstruct -I /home/opam/.opam/4.12/lib/mirage-clock -I /home/opam/.opam/4.12/lib/mirage-clock-unix -I /home/opam/.opam/4.12/lib/mirage-device -I /home/opam/.opam/4.12/lib/mirage-flow -I /home/opam/.opam/4.12/lib/mirage-net -I /home/opam/.opam/4.12/lib/mirage-profile -I /home/opam/.opam/4.12/lib/mirage-protocols -I /home/opam/.opam/4.12/lib/mirage-random -I /home/opam/.opam/4.12/lib/mirage-random-test -I /home/opam/.opam/4.12/lib/mirage-stack -I /home/opam/.opam/4.12/lib/mirage-time -I /home/opam/.opam/4.12/lib/mirage-vnetif -I /home/opam/.opam/4.12/lib/mmap -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocplib-endian -I /home/opam/.opam/4.12/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.12/lib/pcap-format -I /home/opam/.opam/4.12/lib/psq -I /home/opam/.opam/4.12/lib/randomconv -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/rresult -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I src/icmp/.tcpip_icmpv4.objs/byte -I src/ipv4/.tcpip_ipv4.objs/byte -I src/ipv6/.tcpip_ipv6.objs/byte -I src/stack-direct/.tcpip_stack_direct.objs/byte -I src/stack-unix/.icmpv4_socket.objs/byte -I src/stack-unix/.tcp_socket_options.objs/byte -I src/stack-unix/.tcpip_stack_socket.objs/byte -I src/stack-unix/.tcpv4_socket.objs/byte -I src/stack-unix/.tcpv4v6_socket.objs/byte -I src/stack-unix/.tcpv6_socket.objs/byte -I src/stack-unix/.udpv4_socket.objs/byte -I src/stack-unix/.udpv4v6_socket.objs/byte -I src/stack-unix/.udpv6_socket.objs/byte -I src/tcp/.tcp.objs/byte -I src/tcpip_checksum/.tcpip.objs/byte -I src/udp/.tcpip_udpv4.objs/byte -no-alias-deps -open Dune__exe -o test/.test.eobjs/byte/dune__exe__Common.cmo -c -impl test/common.ml)
# File "test/common.ml", line 3, characters 26-39:
# 3 | let failf fmt = Fmt.kstrf Alcotest.fail fmt
#                               ^^^^^^^^^^^^^
# Error: This expression has type
#          (string -> 'a) Alcotest.extra_info =
#            ?here:Alcotest.Source_code_position.here ->
#            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
#        but an expression was expected of type string -> 'b

-       ocamlc test/.test.eobjs/byte/dune__exe__Common.{cmi,cmo,cmt} (exit 2)
- (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.test.eobjs/byte -I /home/opam/.opam/4.12/lib/alcotest -I /home/opam/.opam/4.12/lib/alcotest/engine -I /home/opam/.opam/4.12/lib/arp -I /home/opam/.opam/4.12/lib/arp/mirage -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/cmdliner -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/cstruct-lwt -I /home/opam/.opam/4.12/lib/domain-name -I /home/opam/.opam/4.12/lib/duration -I /home/opam/.opam/4.12/lib/ethernet -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/ipaddr -I /home/opam/.opam/4.12/lib/ipaddr-cstruct -I /home/opam/.opam/4.12/lib/ipaddr/unix -I /home/opam/.opam/4.12/lib/logs -I /home/opam/.opam/4.12/lib/lru -I /home/opam/.opam/4.12/lib/lwt -I /home/opam/.opam/4.12/lib/lwt-dllist -I /home/opam/.opam/4.12/lib/lwt/unix -I /home/opam/.opam/4.12/lib/macaddr -I /home/opam/.opam/4.12/lib/macaddr-cstruct -I /home/opam/.opam/4.12/lib/mirage-clock -I /home/opam/.opam/4.12/lib/mirage-clock-unix -I /home/opam/.opam/4.12/lib/mirage-device -I /home/opam/.opam/4.12/lib/mirage-flow -I /home/opam/.opam/4.12/lib/mirage-net -I /home/opam/.opam/4.12/lib/mirage-profile -I /home/opam/.opam/4.12/lib/mirage-protocols -I /home/opam/.opam/4.12/lib/mirage-random -I /home/opam/.opam/4.12/lib/mirage-random-test -I /home/opam/.opam/4.12/lib/mirage-stack -I /home/opam/.opam/4.12/lib/mirage-time -I /home/opam/.opam/4.12/lib/mirage-vnetif -I /home/opam/.opam/4.12/lib/mmap -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocplib-endian -I /home/opam/.opam/4.12/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.12/lib/pcap-format -I /home/opam/.opam/4.12/lib/psq -I /home/opam/.opam/4.12/lib/randomconv -I /home/opam/.opam/4.12/lib/re -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/rresult -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/uuidm -I /home/opam/.opam/4.12/lib/uutf -I src/icmp/.tcpip_icmpv4.objs/byte -I src/ipv4/.tcpip_ipv4.objs/byte -I src/ipv6/.tcpip_ipv6.objs/byte -I src/stack-direct/.tcpip_stack_direct.objs/byte -I src/stack-unix/.icmpv4_socket.objs/byte -I src/stack-unix/.tcp_socket_options.objs/byte -I src/stack-unix/.tcpip_stack_socket.objs/byte -I src/stack-unix/.tcpv4_socket.objs/byte -I src/stack-unix/.tcpv4v6_socket.objs/byte -I src/stack-unix/.tcpv6_socket.objs/byte -I src/stack-unix/.udpv4_socket.objs/byte -I src/stack-unix/.udpv4v6_socket.objs/byte -I src/stack-unix/.udpv6_socket.objs/byte -I src/tcp/.tcp.objs/byte -I src/tcpip_checksum/.tcpip.objs/byte -I src/udp/.tcpip_udpv4.objs/byte -no-alias-deps -open Dune__exe -o test/.test.eobjs/byte/dune__exe__Common.cmo -c -impl test/common.ml)
- File "test/common.ml", line 3, characters 26-39:
- 3 | let failf fmt = Fmt.kstrf Alcotest.fail fmt
-                               ^^^^^^^^^^^^^
- Error: This expression has type
-          (string -> 'a) Alcotest.extra_info =
-            ?here:Alcotest.Source_code_position.here ->
-            ?pos:Alcotest.Source_code_position.pos -> string -> 'a
-        but an expression was expected of type string -> 'b
```

**Comment by CraigFe** [comment]
@mseri: thanks for picking those out. I've added the necessary bounds and made corresponding patches upstream.

**Comment by mseri** [comment]
I went through all the failures once again. Again they are mostly all succeeded and the few failures have nothing to do with this PR. The tests waiting to run have already run (or timed out) on the previous run. I think we should release the pressure on the CI and merge. Thanks for the bounds and the fixes

**Pull request merged**